### PR TITLE
0.23.4 backports 1

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/22166

### CODE REVIEW

File: pandas/_libs/window.pyx [hunk 1]
```diff
@@ -14,6 +14,7 @@ cnp.import_array()
 
 
 cdef extern from "../src/headers/cmath" namespace "std":
```
Comment (TomAugspurger) [inline pandas/_libs/window.pyx]: @chris-b1 @jbrockmendel this changed between 0.23.x and master. On master we use

```cython
cdef extern from "src/headers/cmath" namespace "std":
```

But here we use `../src/headers/cmath`. Does this look correct for 0.23.x?

Comment (jbrockmendel) [reply pandas/_libs/window.pyx]: This looks like something I would have nit-picked about.  The "../" is inaccurate in this context.  Evidently the cython/gcc compiler doesn't mind either way, but I'm pretty sure the one in master is More Correct.

**Comment by gfyoung** [comment]
> Closes #22166

@TomAugspurger : I thought we were experimenting with a bot that auto-backported?

Separate note: some weird `pytables` failures on Travis and Circle.  Not sure what happened but rerunning to confirm.

**Comment by TomAugspurger** [comment]
> I thought we were experimenting with a bot that auto-backported?

Yes, tagging an issue with a milestone will automatically backport it if possible.

I think the whatsnew changes weren't tagged with the 0.23.4 milestone, because it contained changes that shouldn't be backported.

The window fix had a merge conflict.

**Pull request merged**